# Jupyter Notebook Mod 01: Practice Assignment!

1. Accept the user's first and last name and print them in reverse order with a space between them    

In [1]:
first = input()
last = input()
print(last, first)

 Derek
 Wallace


Wallace Derek


2. Accept an integer (n) input from the user and compute the value of n+nn+nnn. For example if n=4 then n + nn + nnn = 4 + 44 + 444 = 492

In [2]:
n = input()
value = int(n) + int(n + n) + int(n + n + n)
print(value)

 4


492


3. Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!"

In [3]:
print("What country are you from?")
country = input()
print("I have heard that", country, "is a beautiful country!")

What country are you from?


 Sweden


I have heard that Sweden is a beautiful country!


4. What is the output of the following Python code

In [5]:
x = 10
y = 50
if (x ** 2 > 100 and y < 100):
    print(x, y)

There is no output because 10 ** 10 is equal to 100, which is not greater than 100, so the first condition of the conditional fails.

5. What is the output of the following addition (+) operator, and why does this code chunk execute this way?

In [6]:
a = [10, 20]
b = a
b += [30, 40]
print(a)
print(b)

[10, 20, 30, 40]
[10, 20, 30, 40]


The addition operator concatenates the second array to the first, making it \[10, 20, 30, 40\]. Printing both a and b returns the same array because b was assigned to a (b = a), referencing the same underlying data.

6. What is the output of the following code and what arithmetic operators is being used here? print(2%6)

In [7]:
print(2%6)

2


The code outputs "2" because the modulo operator is being used. The statement 2 % 6 essentially evaluates 2 divided by 6 and returns the remainder.

7. What is the output of the following code and what arithmetic operators are used here? print(2 * 3 ** 3 * 4)

In [8]:
print(2*3**3*4)

216


The code outputs "216" using the traditional order of operations for math. The "\*" character is the multiplication operator and the "\*\*" operator is exponent.

8. What is a text editor?

A text editor is any software environment that allows users to edit text files.

9. What is python?

Python is an interpreted programming language that we are learning and eventually going to be using for data science.

10. What is jupyter notebook, what type of python environment is it, and what alternatives are there to jupyter notebook?

Jupyter Notebook is an environment in which all sorts of files (primarily .ipynb in my experience) can be created and edited that we are using to learn Python. The Python notebooks that can be created with it are a combination of Python code and Markdown that allows us to code in Python while simultaneously creating a document for showcasing something. For example, I am using a Jupyter Lab server hosted on my own computer to create this homework document for TTS. Other alternatives would be some sort of Python IDE like PyCharm, VS Code, or just a plain text editor like vim and using the terminal to run the notebook files.